src/pymorize/core/cmorizer.py

```diff
@@ -364,7 +364,30 @@ def _rules_expand_drvs(self):
             else:
                 cloned_rules = rule.expand_drvs()
                 for rule in cloned_rules:
-                    new_rules.append(rule)
+                    # Rule has a table_id or a table_name, so it should only
+                    # match that table
+                    if hasattr(rule, "table_id"):
+                        if isinstance(rule.table_id, str):
+                            rule.table_id = [
+                                rule.table_id,
+                            ]
+                        logger.info(f"Specified table_id as {rule.table_id=}")
+                        for drv in rule.data_request_variables:
+                            if drv.table_header.table_id in rule.table_id:
+                                logger.info(f"Adding rule/table combo for {drv}")
+                                new_rules.append(rule)
+                    elif hasattr(rule, "table_name"):
+                        if isinstance(rule.table_name, str):
+                            rule.table_name = [
+                                rule.table_name,
+                            ]
+                        logger.info(f"Specified table_name as {rule.table_name=}")
+                        for drv in rule.data_request_variables:
+                            if drv.table_header.table_id in rule.table_name:
```

Comment:
Here should be `table_name` right?
```suggestion
                            if drv.table_header.table_name in rule.table_name:
```

Reply:
Sadly no, that would be easy to understand. The `drv` has a thing called `table_header` and that table header only defines the table name as `table_name`. I just gave the users the options to do either one here, the actual code that adds the cloned rules works the same way both times.